servers/quarkus-common/src/main/java/org/projectnessie/quarkus/providers/PersistProvider.java

```diff
@@ -0,0 +1,162 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ApplicationScoped
+@RegisterForReflection(
+    classNames = {
+      "com.github.benmanes.caffeine.cache.SSSMW",
+      "com.github.benmanes.caffeine.cache.PSMW"
+    })
+public class PersistProvider {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PersistProvider.class);
+
+  private final Instance<BackendBuilder> backendBuilder;
+  private final Instance<Backend> backend;
+  private final VersionStoreConfig versionStoreConfig;
+  private final ServerConfig serverConfig;
+  private final QuarkusStoreConfig storeConfig;
+  private final Instance<Tracer> opentelemetryTracer;
+
+  @Inject
+  public PersistProvider(
+      @Any Instance<Tracer> opentelemetryTracer,
+      @Any Instance<BackendBuilder> backendBuilder,
+      @Any Instance<Backend> backend,
+      VersionStoreConfig versionStoreConfig,
+      QuarkusStoreConfig storeConfig,
+      ServerConfig serverConfig) {
+    this.backendBuilder = backendBuilder;
+    this.backend = backend;
+    this.versionStoreConfig = versionStoreConfig;
+    this.storeConfig = storeConfig;
+    this.serverConfig = serverConfig;
+    this.opentelemetryTracer = opentelemetryTracer;
+  }
+
+  @Produces
+  @Singleton
+  @Startup
+  public Backend produceBackend() {
+    VersionStoreType versionStoreType = versionStoreConfig.getVersionStoreType();
+    if (!versionStoreType.isNewStorage()) {
+      return null;
+    }
+
+    if (backendBuilder.isUnsatisfied()) {
+      throw new IllegalStateException("No Quarkus backend implementation for " + versionStoreType);
+    }
+
+    return backendBuilder.select(new Literal(versionStoreType)).get().buildBackend();
+  }
+
+  public void closeBackend(@Disposes Backend backend) throws Exception {
+    if (backend != null) {
+      String info = backend.configInfo();
+      if (!info.isEmpty()) {
+        info = " (" + info + ")";
+      }
+      LOGGER.info("Stopping storage for {}{}", versionStoreConfig.getVersionStoreType(), info);
+      backend.close();
+    }
+  }
+
+  @Produces
+  @Singleton
+  @Startup
+  public Persist producePersist() {
+    VersionStoreType versionStoreType = versionStoreConfig.getVersionStoreType();
+    if (!versionStoreType.isNewStorage()) {
+      return null;
+    }
+
+    if (backend.isUnsatisfied()) {
+      throw new IllegalStateException("No Quarkus backend for " + versionStoreType);
+    }
+
+    Backend b = backend.get();
+    b.setupSchema();
+
+    LOGGER.info("Creating/opening version store {} ...", versionStoreType);
+
+    BiFunction<Persist, String, Persist> wrapPersistTracing = (p, name) -> p;
+    String tracingInfo = "without tracing";
+    if (versionStoreConfig.isTracingEnabled()) {
+      if (opentelemetryTracer.isUnsatisfied()) {
+        LOGGER.warn(
+            "OpenTelemetry is enabled, but not available, forgot to add quarkus-opentelemetry? Will fall back to OpenTracing, if enabled.");
```

Comment:
Not sure I understand... Where is the fallback to OpenTracing happening? Why OpenTracing?